### ISSUE TITLE
kim not installed with correct endpoint

### ISSUE THREAD

**Issue by jandubois** [rancher-sandbox/rancher-desktop#525]
I just installed Rancher Desktop on a freshly updated (but otherwise pristine) macOS Big Sur 11.5.2 instance:

```console
% cat background.log 
app version: 0.4.1
{
  version: 3,
  kubernetes: { version: '', memoryInGB: 4, numberCPUs: 2, port: 6443 },
  portForwarding: { includeKubernetesServices: false },
  images: { showAll: true },
  telemetry: true,
  updater: true
}
k8s-current-port: 6443
Setting k8s integration for /usr/local/bin/kubectl to true
k8s-current-port: 6443
k8s-current-port: 6443
k8s-current-port: 6443
Setting k8s integration for /usr/local/bin/kim to true
```

<details>
<summary>kim.log</summary>

```console
% cat kim.log 
Waiting for Kubernetes node IP to become 192.168.5.15...
Existing kim install invalid: missing endpoint
Installing kim --force
time="2021-08-19T23:18:37-07:00" level=info msg="Applying node-role `builder` to `lima-rancher-desktop`"
time="2021-08-19T23:18:37-07:00" level=info msg="Asserting namespace `kube-image`"
time="2021-08-19T23:18:37-07:00" level=info msg="Asserting TLS secrets"
time="2021-08-19T23:18:38-07:00" level=info msg="Asserting service/endpoints"
time="2021-08-19T23:18:38-07:00" level=info msg="Installing builder daemon"
time="2021-08-19T23:18:38-07:00" level=info msg="Waiting on builder daemon availability..."
time="2021-08-19T23:18:42-07:00" level=warning msg="Cannot find available builder daemon, attempting automatic installation..."
time="2021-08-19T23:18:42-07:00" level=info msg="Applying node-role `builder` to `lima-rancher-desktop`"
time="2021-08-19T23:18:42-07:00" level=info msg="Asserting namespace `kube-image`"
time="2021-08-19T23:18:42-07:00" level=info msg="Asserting TLS secrets"
time="2021-08-19T23:18:42-07:00" level=info msg="Asserting service/endpoints"
time="2021-08-19T23:18:43-07:00" level=info msg="Installing builder daemon"
time="2021-08-19T23:18:43-07:00" level=warning msg="builder already installed"
Error: unknown service port
Usage:
  kim images [OPTIONS] [REPOSITORY[:TAG]]

Aliases:
  images, image list

Flags:
  -a, --all        Show all images (default hides tag-less images)
      --digests    Show digests
  -h, --help       help for images
      --no-trunc   Don't truncate output
  -q, --quiet      Only show image IDs

Global Flags:
  -x, --context string      kubeconfig context for authentication
      --debug               
      --debug-level int     
  -k, --kubeconfig string   kubeconfig for authentication
  -n, --namespace string    namespace (default "kube-image")

time="2021-08-19T23:18:43-07:00" level=fatal msg="unknown service port"

time="2021-08-19T23:18:42-07:00" level=warning msg="Cannot find available builder daemon, attempting automatic installation..."
time="2021-08-19T23:18:42-07:00" level=info msg="Applying node-role `builder` to `lima-rancher-desktop`"
time="2021-08-19T23:18:42-07:00" level=info msg="Asserting namespace `kube-image`"
time="2021-08-19T23:18:42-07:00" level=info msg="Asserting TLS secrets"
time="2021-08-19T23:18:42-07:00" level=info msg="Asserting service/endpoints"
time="2021-08-19T23:18:43-07:00" level=info msg="Installing builder daemon"
time="2021-08-19T23:18:43-07:00" level=warning msg="builder already installed"
Error: unknown service port
Usage:
  kim images [OPTIONS] [REPOSITORY[:TAG]]

Aliases:
  images, image list

Flags:
  -a, --all        Show all images (default hides tag-less images)
      --digests    Show digests
  -h, --help       help for images
      --no-trunc   Don't truncate output
  -q, --quiet      Only show image IDs

Global Flags:
  -x, --context string      kubeconfig context for authentication
      --debug               
      --debug-level int     
  -k, --kubeconfig string   kubeconfig for authentication
  -n, --namespace string    namespace (default "kube-image")

time="2021-08-19T23:18:43-07:00" level=fatal msg="unknown service port"

time="2021-08-19T23:18:44-07:00" level=info msg="Waiting on builder daemon availability..."
kim images: Error: unknown service port #2
time="2021-08-19T23:18:50-07:00" level=info msg="Waiting on builder daemon availability..."
kim images: Error: unknown service port #3
time="2021-08-19T23:18:57-07:00" level=info msg="Waiting on builder daemon availability..."
kim images: Error: unknown service port #4
time="2021-08-19T23:19:03-07:00" level=info msg="Waiting on builder daemon availability..."
kim images: Error: unknown service port #5
kim images: Error: unknown service port #6
time="2021-08-19T23:19:08-07:00" level=info msg="Waiting on builder daemon availability..."
kim images: Error: unknown service port #7
time="2021-08-19T23:18:37-07:00" level=warning msg="Cannot find available builder daemon, attempting automatic installation..."
time="2021-08-19T23:18:37-07:00" level=info msg="Applying node-role `builder` to `lima-rancher-desktop`"
time="2021-08-19T23:18:37-07:00" level=info msg="Asserting namespace `kube-image`"
time="2021-08-19T23:18:37-07:00" level=info msg="Asserting TLS secrets"
time="2021-08-19T23:18:38-07:00" level=info msg="Asserting service/endpoints"
time="2021-08-19T23:18:38-07:00" level=info msg="Installing builder daemon"
time="2021-08-19T23:18:38-07:00" level=info msg="Waiting on builder daemon availability..."
time="2021-08-19T23:18:44-07:00" level=info msg="Waiting on builder daemon availability..."
time="2021-08-19T23:18:49-07:00" level=info msg="Waiting on builder daemon availability..."
time="2021-08-19T23:18:54-07:00" level=info msg="Waiting on builder daemon availability..."
time="2021-08-19T23:19:00-07:00" level=info msg="Waiting on builder daemon availability..."
time="2021-08-19T23:19:05-07:00" level=info msg="Waiting on builder daemon availability..."
time="2021-08-19T23:19:11-07:00" level=info msg="Waiting on builder daemon availability..."
Error: rpc error: code = Unavailable desc = connection error: desc = "transport: authentication handshake failed: x509: certificate is valid for 192.168.5.15, not 127.0.0.1"
Usage:
  kim images [OPTIONS] [REPOSITORY[:TAG]]

Aliases:
  images, image list

Flags:
  -a, --all        Show all images (default hides tag-less images)
      --digests    Show digests
  -h, --help       help for images
      --no-trunc   Don't truncate output
  -q, --quiet      Only show image IDs

Global Flags:
  -x, --context string      kubeconfig context for authentication
      --debug               
      --debug-level int     
  -k, --kubeconfig string   kubeconfig for authentication
  -n, --namespace string    namespace (default "kube-image")

time="2021-08-19T23:19:17-07:00" level=fatal msg="rpc error: code = Unavailable desc = connection error: desc = \"transport: authentication handshake failed: x509: certificate is valid for 192.168.5.15, not 127.0.0.1\""

Error: rpc error: code = Unavailable desc = connection error: desc = "transport: authentication handshake failed: x509: certificate is valid for 192.168.5.15, not 127.0.0.1"
Usage:
  kim images [OPTIONS] [REPOSITORY[:TAG]]

Aliases:
  images, image list

Flags:
  -a, --all        Show all images (default hides tag-less images)
      --digests    Show digests
  -h, --help       help for images
      --no-trunc   Don't truncate output
  -q, --quiet      Only show image IDs

Global Flags:
  -x, --context string      kubeconfig context for authentication
      --debug               
      --debug-level int     
  -k, --kubeconfig string   kubeconfig for authentication
  -n, --namespace string    namespace (default "kube-image")

time="2021-08-19T23:19:17-07:00" level=fatal msg="rpc error: code = Unavailable desc = connection error: desc = \"transport: authentication handshake failed: x509: certificate is valid for 192.168.5.15, not 127.0.0.1\""

kim images: Error: rpc error: code = Unavailable desc = connection error: desc = "transport: authentication handshake failed: x509: certificate is valid for 192.168.5.15, not 127.0.0.1" #2
kim images: Error: rpc error: code = Unavailable desc = connection error: desc = "transport: authentication handshake failed: x509: certificate is valid for 192.168.5.15, not 127.0.0.1" #3
kim images: Error: rpc error: code = Unavailable desc = connection error: desc = "transport: authentication handshake failed: x509: certificate is valid for 192.168.5.15, not 127.0.0.1" #4
[...]
```
</details>

**Comment by mook-as** [comment]
`kim.log` ends with: `kim images: Error: rpc error: code = Unavailable desc = connection error: desc = "transport: authentication handshake failed: x509: certificate is valid for 192.168.5.15, not 127.0.0.1" #4`

So we did tell it to use the correct endpoing (`127.0.0.1`), but the cert wasn't regenerated correctly.  We did tell kim to redo things (`Installing kim --force`), though; maybe we should double check if we're checking the cert correctly…

**Comment by jandubois** [comment]
I have not been able to reproduce this issue. I've started over with the same pristine Big Sur 11.5.2 image, installed RD 0.4.1, and kim worked correctly. I stopped and restarted RD, and kim continued to work. Then I rebooted the VM, and RD came back up automatically, and kim still worked fine.

I can't think of anything else, except wait to see if this will happen again sometime.

**Comment by gaktive** [comment]
Unable to repro. Please file a new issue if this recurs.
